internal/provider/kubernetes/predicates.go

```diff
@@ -168,7 +168,20 @@ func (r *gatewayAPIReconciler) validateServiceForReconcile(obj client.Object) bo
 		r.log.Info("unexpected object type, bypassing reconciliation", "object", obj)
 		return false
 	}
+	labels := svc.GetLabels()
+	gclass, ok := labels[gatewayapi.OwningGatewayClassLabel]
+	if ok {
+		gtw := r.findGateways(ctx, gclass)
+		if len(gtw.Items) != 0 {
+			for _, gw := range gtw.Items {
+				gw := gw
+				r.statusUpdateForGateway(ctx, &gw)
+			}
+		}
 
+		nsName := utils.NamespacedName(svc)
```

Comment:
curious why this (backend svc) is needed for this case (where the `svc` here is the generated envoy svc)?

Reply:
we should be returning `false` here

Reply:
good catch